pipenv/environment.py

```diff
@@ -712,6 +713,23 @@ def is_installed(self, pkgname):
 
         return any(d for d in self.get_distributions() if d.project_name == pkgname)
 
+    def is_satisfied(self, req):
+        match = next(
+            iter(
+                d for d in self.get_distributions()
+                if canonicalize_name(d.project_name) == req.normalized_name
+            ), None
+        )
+        if match is not None:
+            if req.editable and self.find_egg(match):
+                return req.line_instance.path == match.location
+            elif req.line_instance.specifiers is not None:
+                return req.line_instance.specifiers.contains(
+                    match.version, prereleases=True
+                )
+            return True
```

Comment:
Probably need to add one more branch to check direct URL install.

Reply:
hmmm yes, I assumed they would just surface the version here. thanks!